File: backend/Domain/Model/HotelBooking.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Model
{
    public class HotelBooking
    {
        public Guid HotelBookingId { get; set; }
        public Guid userID { get; set; }
        public virtual User user { get; set; }
        public Guid HotelID { get; set; }
        public virtual Hotel hotel { get; set; }
        public DateTime BookingDate { get; set; }
        public DateTime Checkindate { get; set; }
        public DateTime Checkoutdate { get; set; } 
        public string roomType { get; set; }
        public int noofPeople { get; set; }
        public string bookingStatus { get; set; } 
        public int Price { get; set; }
    }
}

File: frontend/src/api/hotelBooking.ts

import axios from "axios";

const BASE_URL = "https://localhost:7032/api/HotelBooking";


export const getHotelBookings = () => {
    return axios.get(`${BASE_URL}/GetAllHotelBooking`);
};

// GET a booking by ID
export const getHotelBookingById = (id: string) => {
    return axios.get(`${BASE_URL}/GetHotelBookingByID`, {
        params: { id },
    });
};

// ADD a booking
export const addHotelBooking = (data: {
    userID: string;
    HotelID: string;
    Checkindate: string;
    Checkoutdate: string;
    roomType: string;
    BookingDate: string;
    noofPeople: string;
    bookingStatus: string,
    price: string
}) => {
    return axios.post(`${BASE_URL}/AddHotelBooking`, data);
};

export const editHotelBooking = (
    id: string,
    data: {
        HotelBookingId: string;
        userID: string;
        HotelID: string;
        Checkindate: string;
        Checkoutdate: string;
        roomType: string;
        BookingDate: string;
        noofPeople: string;
        bookingStatus: string,
        price: string
    }
) => {
    const payload = {
        ...data,
        HotelBookingId: id,
    };
    return axios.patch(`${BASE_URL}/EditHotelBooking/${id}`, payload);
};

// DELETE a booking by ID
export const deleteHotelBooking = (id: string) => {
    return axios.delete(`${BASE_URL}/RemoveHotelBooking/${id}`);
};
